src/graph_compute/io.cpp

```diff
@@ -94,3 +95,73 @@ int readEdgesFromFile(const string filepath,
 
   return edgelistfile_read(filepath, &builder);
 }
+
+void makeSimpleAndUndirected(vertex_t * nodes,
```

Comment:
Can we change it to cause an error or something like that if the graph is not simple / doesn't satisfy any of the constraints? I don't like the idea of silently reconfiguring the graph, it feels like suppressing the error rather than telling the user what's wrong.